### DISCUSSION TITLE
Using Mask R-CNN for Object Detection

### DISCUSSION THREAD

**Discussion by brookconstantz** [torchgeo/torchgeo#2355] [Q&A]
I noticed through several articles that TorchGeo uses a Fast R-CNN with their ObjectDetection task. Is it possible to use something like Mask R-CNN with this task instead? I would really like to try out Mask R-CNN for my habitat mapping effort. Additionally, I'm also trying to get a good scope of all of the types of tasks and models compatible with TorchGeo, as I don't see all of them listed directly in the models or task pages on the Read the Docs website.

Thanks!

**Comment by adamjstewart** [comment]
> I noticed through several articles that TorchGeo uses a Fast R-CNN with their ObjectDetection task.

We currently support Faster R-CNN, FCOS, and RetinaNet

> Is it possible to use something like Mask R-CNN with this task instead?

Definitely! Torchvision has Mask R-CNN, so we just need to add it as an option in [detection.py](https://github.com/microsoft/torchgeo/blob/main/torchgeo/trainers/detection.py). Want to open a PR?

> I'm also trying to get a good scope of all of the types of tasks and models compatible with TorchGeo, as I don't see all of them listed directly in the models or task pages on the Read the Docs website.

Types of tasks can be found [here](https://torchgeo.readthedocs.io/en/stable/api/trainers.html). Types of models are everything in [timm](https://github.com/huggingface/pytorch-image-models) (700+ models), [SMP](https://github.com/qubvel-org/segmentation_models.pytorch), some models from [torchvision](https://github.com/pytorch/vision), and any models listed [here](https://torchgeo.readthedocs.io/en/stable/api/models.html).

Of course, you don't _have_ to use our PyTorch Lightning trainers, you could use our data loaders and combine them with any model ever written, even ones for TensorFlow.
